opentracing-usecases/src/main/java/io/opentracing/usecases/AutoFinishScopeManager.java

```diff
@@ -0,0 +1,64 @@
+package io.opentracing.usecases;
+
+import io.opentracing.Scope;
+import io.opentracing.ScopeManager;
+import io.opentracing.Span;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Created by bhs on 8/5/17.
+ */
+public class AutoFinishScopeManager implements ScopeManager {
```

Comment:
This class demonstrates that the refcounting can be built entirely outside of the core OT API